### ISSUE TITLE
Foreign Key Index/Constraint Not Created on Updated Table

### ISSUE THREAD

**Issue by jason-roberts** [JasperFx/marten#590]
This may have already been fixed in #553 but just wanted to mention here in case it's a bug.

Steps to repro:
1. Create db
2. Create class A with properties but no foreign key (FK) property yet
3. Store doc instance in db
4. Add second class B
5. Modify A: add property (FK) to reference doc A Id (to be used as FK)
6. Either use [ForeignKey] attribute in A or configure with DocumentStore.For ... Schema.For
7. In a session, store instance of B, set new instance of A's FK prop to B's Id
8. On SaveChanges, table B created ok, table A updated with new column but **no foreign key constraint added** to table A
9. This means that referential integrity is not enforced between A & B
- Clearing all data from table A before executing step 7 makes no difference
- Using attribute or Schema.For makes no difference 
- If table A is manually dropped before executing step 7, table A is recreated successfully with correct foreign key constraint
- Makes no difference if FK prop in A is nullable or not

Marten 1.1 NuGet
PostgreSQL 9.6 Windows

**Comment by jeremydmiller** [comment]
Any chance I could get you to do a PR with a failing test for this?

**Comment by jason-roberts** [comment]
I started trying to write test for this but needed to modify definition of class after 1st insert so was unsure how to do this and then configure in Marten. I've thrown together this to demo: https://github.com/jason-roberts/MartenFKDemo run first console app first, then second console app. Assumes host = localhost; database = MartenSpike; username = postgres" - you'll need to edit password/cn string in both apps before running.

On running 2nd app you should get "SaveChanges should have thrown a foreign key constraint violation exception but did not". 

Resulting tables in pgAdmin:

![untitled](https://cloud.githubusercontent.com/assets/1472335/19795829/efc0df56-9d11-11e6-8b07-d3c4cd3f1b54.png)

Hope that helps :)

**Comment by jeremydmiller** [comment]
I know what's (not) happening now. The cute little "generate a patch" code doesn't consider constraints whatsoever. I'll add that in a bit and get the fix rolled into 1.2 later this week.

Thanks for doing that. I've got a failing UT for it now:

```
    public class Bug_590_add_FK_later
    {
        [Fact]
        public void should_add_a_new_FK_to_the_database()
        {
            using (var store = DocumentStore.For(_ => _.Connection(ConnectionSource.ConnectionString)))
            {
                store.Advanced.Clean.CompletelyRemoveAll();
                store.Schema.EnsureStorageExists(typeof(UserHolder));
            }

            using (var store = DocumentStore.For(_ =>
            {
                _.Connection(ConnectionSource.ConnectionString);
                _.Schema.For<UserHolder>().ForeignKey<User>(x => x.UserId);
            }))
            {
                store.Schema.EnsureStorageExists(typeof(UserHolder));

                store.Schema.DbObjects
                    .AllForeignKeys()
                    .Any(x => x.Name == "mt_doc_userholder_user_id_fkey")
                    .ShouldBeTrue();
            }
        }
    }

    public class UserHolder
    {
        public Guid Id;

        public Guid UserId { get; set; }
    }
```
